pennylane_qiskit/qiskit_device2.py

```diff
@@ -372,8 +369,34 @@ def preprocess(
 
         return transform_program, config
 
-    @staticmethod
-    def get_transpile_args(kwargs):
+    def _process_kwargs(self):
+        """Combine the settings defined in options and the settings passed as kwargs, with
+        the definition in options taking precedence if there is conflicting information"""
+
+        if "noise_model" in self._kwargs:
+            noise_model = self._kwargs.pop("noise_model")
+            self.backend.set_options(noise_model=noise_model)
+
+        if "options" in self._kwargs:
+            for key, val in self._kwargs.pop("options").items():
+                if key in self._kwargs:
+                    warnings.warn(
+                        "An overlap between what was passed in via options and what was passed in via kwargs was found."
+                        f"The value set in options {key}: {val} will be used."
+                    )
+                self._kwargs[key] = val
+
+        shots = self._kwargs.pop("shots")
+
+        if "default_shots" in self._kwargs:
+            warnings.warn(
+                f"default_shots was found as a keyword argument, but it is not supported by {self.name}"
+                "Please use the `shots` keyword argument instead. The number of shots "
+                f"{shots} will be used instead."
+            )
+        self._kwargs["default_shots"] = shots
+
+    def get_transpile_args(self):
```

Comment:
Changed the signature of this function because we want to `pop` the kwargs that are going to be passed to the transpiler. Otherwise when these kwargs reach the primitive, it'll raise a `ValidationError`.

Reply:
What happens if you run this on two consecutive circuits? Does the second one miss the `transpile` arguments because they've been removed?

Maybe this should be called once at initialization and store all the arguments as a `self._transpile_args` property instead of calling this each time we compile.